44_質問集ページ見た目調整

Ikko-T opened the pull request:
## 概要
質問投稿及び質問詳細画面の見た目調整

## 実装内容
- 「送信」ボタンの色とサイズ変更
- 「h1」と「h3」テキストの文字の大きさと位置調整
-  入力フォームの大きさ、形状、（divタグを用いて）位置調整

## 参考資料
- やんばるCODE教材: Bootstrap教材（その1 & 2）／HTML&CSS 実践編1, 2
- https://webst8.com/blog/bootstrap4-display/

## コメント
元アプリと完全に同じ見た目ではありません。
- 新規投稿質問一覧では余白が設定されていません。一応、paddingやmarginなど色々試しましたが出来ませんでした。そこまで見た目的に問題はないであろうと最終的に判断しましたが、必要実装でしたらご指摘お願いします！
- 質問詳細画面の各項目に表示される棒線も上記同様理由。

Ikko-T committed:
Aligned text, form, and button in question in css.

Ikko-T committed:
Aligned size of form and button.

Ikko-T committed:
Aligned the size of text, form, and button in question detail.

Ikko-T committed:
Fixed conflicts.

T-Tsujii commented on app/views/questions/index.html.erb:
```diff
@@ -1,22 +1,24 @@
-  <div>
-    <%= form.label :title, "質問" %>
-    <%= form.text_area :title, cols: 50, row: 3, size: 8 %>
+  <div class="question-title">
```
Bootstrapを使用したフォームの書き方を確認していただけますか。
https://getbootstrap.jp/docs/4.5/components/forms/

`row`を指定するのは問題ありませんが，`cols` や `size` は指定すべきものではありません。

Ikko-T replied:
`row`も削除し、`form-control`クラス名を用いました。

T-Tsujii commented on app/views/questions/index.html.erb:
```diff
@@ -1,22 +1,24 @@
   <% @questions.each do |question| %>
     <%= link_to question do %>
-      <div class="row question">
+      <div class="row question border rounded">
```
スタイル部分が難しいと思いますが，元アプリではこのようなスタイルがついています。

```suggestion
      <div class="row question">
```

```scss
.question-container {
  padding: 30px 0;

  a {
    text-decoration: none;
    display: block;
    color: #333333;
    border: solid 2px #505050;
    border-radius: 10px;
    transition: .4s;

    .title-container {
      word-wrap: normal;
    }

    .title {
      font-size: 22px;
      line-height: 2;
    }

    .info {
      padding: 10px 10px 10px 20px;
      color: #BBBBBB;
    }

    :hover {
      border: solid 2px #7d9ccb;
      -webkit-transform: translateY(-5px);
      -ms-transform: translateY(-5px);
      transform: translateY(-5px);
      transition: .4s;
    }
  }

  .question {
    padding: 10px;
  }
}
```

Ikko-T replied:
上記コピペで適用しました。

Ikko-T committed:
Improved layouts used in HTML.erb & CSS.

T-Tsujii commented on app/javascript/stylesheets/application.scss:
```diff
@@ -65,6 +65,33 @@ body {
   height: 250px;
 }
 
+td a:hover{
+  cursor:pointer;
+  color:#34569e !important;
+}
+
+h1 {
+  padding: 20px 0px 10px 0px;
+  font-size: 2rem;
+  text-align: center;
+}
+
+h3 {
+  padding: 20px 0px 10px 0px;
+  font-size: 1rem;
+  text-align: left;
+}
```
このような影響範囲の広いスタイルの付け方は避けて下さい。
他のページにも影響がでる可能性が高くなります。

```suggestion
```

Ikko-T replied:
上記の`td a:hover`は重複していましたので、一つ削除しました。
`h1`と `h3`タグは残し、cssファイルの上部に移動しました。

T-Tsujii commented on app/javascript/stylesheets/application.scss:
```diff
@@ -65,6 +65,33 @@ body {
+  padding: 10px;
+}
+
 td a:hover {
```
```
td a:hover {
  cursor: pointer;
  color: #34569e !important;
}
h1 {
  margin: 10px auto;
  max-width: 300px;
}
```

気づかなかったのですが，このあたりのスタイルも影響範囲が大きすぎるのでよろしくないですね。
別のプルリクでも良いですが，早めに削った方がよいです。

Ikko-T replied:
上記は、すでに記述されていました。リモートブランチにpusした際にコンフリクトが起きたので、両方取り入れました。
ただ、上記の`h1`は削除しました。

T-Tsujii commented on app/javascript/stylesheets/application.scss:
```diff
@@ -65,6 +65,33 @@ body {
+  text-align: left;
+}
+
+.question-title {
+  margin: 10px;
+  padding: 10px;
+}
+
+.question-detail {
+  margin: 10px;
+  padding: 10px;
+}
```
このスタイルは `form-group` を適切に入れていれば必要ないかもしれません。少なくとも `question-container` の中に入れるべきです。

Ikko-T replied:
`form-group`にまとめ、`question-container`の中に入れました。

T-Tsujii commented on app/javascript/stylesheets/application.scss:
```diff
@@ -78,4 +105,48 @@ h1 {
+  }
+}
+
+.form-control {
+  margin-bottom: 10px;
+}
```
これも影響範囲が大きすぎるので避けるべきです。

```suggestion
```

Ikko-T replied:
`.form-control`は`question-container`の中に移動しました。

Ikko-T replied:
`.form-control`はcssから削除しました。

T-Tsujii commented on app/views/questions/index.html.erb:
```diff
@@ -1,16 +1,18 @@
+    <%= form.text_area :title, class:'form-control' %>
+   </div>
   </div>
-  <div>
-    <%= form.label :detail, "詳細" %>
-    <%= form.text_area :detail, cols: 50, row: 6, size: 8 %>
-  </div>
-  <div>
-    <%= form.submit "送信" %>
+  <div class="question-detail">
+    <%= form.label :detail, "【詳細】" %>
+    <div>
+      <%= form.text_area :title, class:'form-control' %>
+   </div>
   </div>
```
`form-control` だけでなく `form-group` も必要です。また，ボタンには `btn` クラスも必要です。
https://getbootstrap.jp/docs/4.5/components/forms/

```erb
<%= form_with model: @question, local: true do |form| %>
  <div class="form-group">
    <%= form.label :title, "【質問】" %>
    <%= form.text_area :title, class: 'form-control' %>
  </div>
  <div class="form-group">
    <%= form.label :detail, "【詳細】" %>
    <%= form.text_area :title, class: 'form-control' %>
  </div>
  <%= form.submit "送信", class: 'btn btn-block btn-info' %>
<% end %>
```

Ikko-T replied:
`question-title`と`question-detail`両方のclass名を削除し、`form-group`の中に入れました。

Ikko-T replied:
`btn`クラスも追加しました。

T-Tsujii commented on app/javascript/stylesheets/application.scss:
```diff
@@ -78,4 +105,48 @@ h1 {
+      color: #BBBBBB;
+    }
+
+    :hover {
```
すみません，`&`の付け忘れです。
```suggestion
        &:hover {
```

Ikko-T replied:
ありがとうございます。

Ikko-T committed:
Removed one of each duplicated tag in CSS.
Moved 'h1', 'h3', and 'body' tag up to top in CSS.
Changed '.question-title' & '.question-detail' to form-group class.
Removed `.form-control` from CSS.
Added a 'btn' to the button class.

Ikko-T committed:
Created a module scss specific to the question page.

Ikko-T committed:
Removed the form-control and btn class in scss.  Aligned the 'a' tag in question-container in scss.

T-Tsujii reviewed (approved):
LGTMです！
